libs/area_reporting.py

```diff
@@ -0,0 +1,72 @@
+import os
+import time
+import logging
+import csv
+from datetime import date
+from collections import deque
+from .utils.mailing import MailService
+from .notifications.slack_notifications import SlackService
+
+logger = logging.getLogger(__name__)
+
+
+class AreaReporting:
+
+    def __init__(self, config, area):
+        self.processing_alerts = False
+        self.config = config
+        self.area = area
+
+        self.occupancy_sleep_time_interval = float(self.config.get_section_dict("App")["OccupancyAlertsTimeout"])
+        self.log_dir = self.config.get_section_dict("Logger")["LogDirectory"]
+        self.idle_time = float(self.config.get_section_dict('Logger')['TimeInterval']) + 0.1
+        self.area_id = self.area['id']
+        self.area_name = self.area['name']
+        self.occupancy_threshold = self.area['occupancy_threshold']
+        self.emails = self.area['emails']
+        self.should_send_email_notifications = self.area['should_send_email_notifications']
+        self.should_send_slack_notifications = self.area['should_send_slack_notifications']
+        self.cameras = [camera for camera in self.config.get_video_sources() if camera['id'] in self.area['cameras']]
+        for camera in self.cameras:
+            camera['file_path'] = os.path.join(self.log_dir, camera['id'], "objects_log")
+            camera['last_processed_time'] = time.time()
+
+        self.mail_service = MailService(config)
+        self.slack_service = SlackService(config)
+
+    def process_area(self):
+        # Sleep for a while so cameras start processing
+        time.sleep(30)
+
+        self.processing_alerts = True
+        logger.info(f'Enabled processing alerts for - {self.area_id}: {self.area_name} with {len(self.cameras)} cameras')
+        while self.cameras and self.processing_alerts:
```

Comment:
Why we need the `self.cameras` condition in the while true?
Taking into accout that all the areas should have unless a camera this condition will always be `True`

Reply:
Well technically one could edit the config.ini manually, I think it checking this "just in case" is not a bad idea. Although probably some other things break without this.

I'd still leave it until we validate the config files before starting up the processor

Reply:
Ok, I that case maybe we can add the condition
```
if not self.cameras:
   return
```
It sounds weird to me a `while` condition that will not change in the middle of the execution.
What do you think?

Reply:
I ended up adding it so that it doesn't create the thread if the area has no cameras